airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/operation/DefaultCheckOperation.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core.operation
+
+import io.airbyte.cdk.core.context.env.ConnectorConfigurationPropertySource
+import io.airbyte.cdk.core.operation.executor.OperationExecutor
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Requires
+import jakarta.inject.Named
+import jakarta.inject.Singleton
+
+private val logger = KotlinLogging.logger {}
+
+@Singleton
+@Named("checkOperation")
+@Requires(
+    property = ConnectorConfigurationPropertySource.CONNECTOR_OPERATION,
+    value = "check",
+)
+class DefaultCheckOperation(
+    @Named("checkOperationExecutor") private val operationExecutor: OperationExecutor,
+) : Operation {
+    override fun type(): OperationType {
+        return OperationType.CHECK
+    }
+
+    override fun execute(): Result<AirbyteMessage?> {
+        logger.info { "Using default check operation." }
+        val result = operationExecutor.execute()
+        result.onFailure {
+            return Result.success(
+                AirbyteMessage()
+                    .withType(AirbyteMessage.Type.CONNECTION_STATUS)
+                    .withConnectionStatus(
+                        AirbyteConnectionStatus()
+                            .withStatus(AirbyteConnectionStatus.Status.FAILED)
+                            .withMessage(it.message),
+                    ),
+            )
+        }
+        return result
+    }
```

Comment:
Doesn't this already get done in `IntegrationCommand`?

Reply:
@postamar No.  The `IntegrationCommand` calls this class, which actually executes the specific logic for the command.  Admittedly, there may be an extra layer here, but I think that we should solve that if/when we separate the types of commands to support the different return types.